src/django_components/slots.py

```diff
@@ -1667,3 +1671,24 @@ def render_func(ctx: SlotContext) -> SlotResult:
 
 def _is_extracting_fill(context: Context) -> bool:
     return context.get(FILL_GEN_CONTEXT_KEY, None) is not None
+
+
+# Fix for compatibility with Django's `{% include %}` and `{% extends %}` tags.
+# See https://github.com/django-components/django-components/issues/1325
+#
+# When we search for `{% fill %}` tags, we also evaluate `{% include %}` and `{% extends %}`
+# tags if they are within component body (between `{% component %}` / `{% endcomponent %}` tags).
+# But by doing so, we trigger Django's block/extends logic to remember that this extended file
+# was already walked.
+# (See https://github.com/django/django/blob/0bff53b4138d8c6009e9040dbb8916a1271a68d7/django/template/loader_tags.py#L114)  # noqa: E501
+#
+# We need to clear that state, otherwise Django won't render the extended template the second time
+# (when we actually render it).
+@contextmanager
+def _extends_context_reset(context: Context) -> Generator[None, None, None]:
```

Comment:
This is the actual fix - not much to it. I extracted it into its own function so it doens't pollute.